cinema/views.py

```diff
@@ -97,27 +86,42 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
-        if self.action == "retrieve":
+        elif self.action == "retrieve":
             return MovieDetailSerializer
+        elif self.action == "upload_image":
+            return MovieImageSerializer
 
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        url_path="upload-image",
+        permission_classes=[IsAdminUser],
+    )
+    def upload_image(self, request: Request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
```

Comment:
You don't need `if` statement if you pass `raise_exception=True`
```suggestion
        serializer.is_valid(raise_exception=True):
        serializer.save()
```